### PR TITLE
Michigan Homestead Property Tax Credit 

### LINKED ISSUE #2350: Michigan Homestead Property Tax Credit
Homestead Property Tax Credit Information
https://www.michigan.gov/taxes/property/ind-exempt-defer/homestead-property-tax-credit-information

2022 Michigan Homestead Property Tax Credit Claim MI-1040CR
https://www.michigan.gov/taxes/-/media/Project/Websites/taxes/Forms/2022/2022-IIT-Forms/MI-1040CR.pdf?rev=3960a0f9b1de4f999acf1bb4209fc8dc&hash=956AD0168F04E84CF9841653AF0523A0

Tax Form Instruction
https://www.michigan.gov/taxes/-/media/Project/Websites/taxes/Forms/2022/2022-IIT-Forms/MI-1040CR-Instructions.pdf?rev=67deea154fe14149a8a22a30e370fa6c&hash=D1B4A597CDE814D95EC9F1DDC91663D3

HOMESTEAD PROPERTY TAX CREDIT Maximum Amount
https://michiganfreetaxhelp.org/tax-credits/#:~:text=The%20Homestead%20Property%20Tax%20Credit,must%20be%20%24143%2C000%20or%20less.

### PR DESCRIPTION
**Pull request by jaewoongy**
### Summary
📝📄🧮

This pull request adds the homestead property tax credit for Michigan to the policy engine. It creates new parameter and variable files to define and calculate the credit based on household resources and property taxes.

> _`homestead_property`_
> _New files and parameters_
> _Autumn tax credit_

### Walkthrough
*  Add a new variable `mi_homestead_property_tax_credit` to calculate the amount of the homestead property tax credit for each tax unit in Michigan ([link](https://github.com/PolicyEngine/policyengine-us/pull/2352/files?diff=unified&w=0#diff-7eb9d8b80412f82a5d42a4a461c9c579a33bc19cb53825e5664bfda9124cfb06L1-R18))
* Add new parameter files `max_amount.yaml` and `phase_out.yaml` to define the maximum amount and the phase out schedule of the homestead property tax credit ([link](https://github.com/PolicyEngine/policyengine-us/pull/2352/files?diff=unified&w=0#diff-4d59d9f0268ff6f3e5c283212a6f720be2fb0d51426f047cac9cb0dc087d3cf1L1-R9), [link](https://github.com/PolicyEngine/policyengine-us/pull/2352/files?diff=unified&w=0#diff-3b4de13b4d7af9b55afc046071eb7a04402d9c255278835a2ac31d327b0a6820L1-R53))
* Add a comment to the `household_resources.yaml` file to specify the unit and the list of variables that are included in the household resources definition for the homestead property tax credit ([link](https://github.com/PolicyEngine/policyengine-us/pull/2352/files?diff=unified&w=0#diff-838788a89c21fcc0975fb431ed820c1bf05347ce187869f3a7d708b0b8d3045dR1-R2))

fixes #2350

### CODE REVIEW

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+- bump: minor
+  changes:
+    added:
+    - Michigan household resources
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
```

File: changelog_entry.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+    - Michigan homstead property tax credit
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:5]:
```suggestion
    - Michigan homstead property tax credit.
```

File: policyengine_us/parameters/gov/states/mi/tax/income/credits/homestead_property_tax_credit/household_resources.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# Source: On page 2 of https://www.michigan.gov/taxes/-/media/Project/Websites/taxes/Forms/2022/2022-IIT-Forms/MI-1040CR7.pdf?rev=c36e54deae424566bb6abd1cbe8af7cb&hash=B39BA9887DD31395E754CC0641EFB512
+
+description: Income sources used to calculate total house resources for the homestead property tax credit
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mi/tax/income/credits/homestead_property_tax_credit/household_resources.yaml:3]:
```suggestion
description: Michigan allows for the following recourses to be utilized for the homestead property tax credit calculation.
```

File: policyengine_us/parameters/gov/states/mi/tax/income/credits/homestead_property_tax_credit/household_resources.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    - rental_income #22
+    - farm_rent_income #22
+    - pension_income #23
+    - net_capital_gain #24
+    - alimony_income #25
+    - social_security #26
+    - ssi #26
+    - child_support_received #27
+    - unemployment_compensation #28
+    - # Missing 29 Gifts received or expenses paid on your behalf
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mi/tax/income/credits/homestead_property_tax_credit/household_resources.yaml:22]:
```suggestion
    - # Gifts received or 
    - # Expenses paid on your behalf
```

File: policyengine_us/parameters/gov/states/mi/tax/income/credits/homestead_property_tax_credit/household_resources.yaml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    - dividend_income #20
+    - interest_income #20
+    - farm_income #21
+    - qualified_business_income #21?
+    - rental_income #22
+    - farm_rent_income #22
+    - pension_income #23
+    - net_capital_gain #24
+    - alimony_income #25
+    - social_security #26
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mi/tax/income/credits/homestead_property_tax_credit/household_resources.yaml:18]:
```suggestion
    - social_security
```

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/mi/tax/income/credits/homestead_property_tax_credit/household_resources.yaml:18]: and others

File: policyengine_us/variables/gov/states/mi/tax/income/credits/homestead_property_tax_credit/mi_household_resources.py [hunk 6]
```diff
@@ -0,0 +1,12 @@
+from policyengine_us.model_api import *
+
+# policyengine-core test ./policyengine_us/tests/policy/baseline/gov/states/mi/
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mi/tax/income/credits/homestead_property_tax_credit/mi_household_resources.py:3]:
```suggestion
```

File: policyengine_us/variables/gov/states/mi/tax/income/credits/homestead_property_tax_credit/mi_homestead_property_tax_credit.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
+from policyengine_us.model_api import *
+
+# policyengine-core test ./policyengine_us/tests/policy/baseline/gov/states/mi/
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mi/tax/income/credits/homestead_property_tax_credit/mi_homestead_property_tax_credit.py:3]:
```suggestion
```

**Review by PavelMakarchuk — COMMENTED** [review]
Comment

File: policyengine_us/parameters/gov/states/mi/tax/income/credits/homestead_property_tax_credit/max_amount.yaml [hunk 8]
```diff
@@ -0,0 +1,10 @@
+description: Michigan provides this maximum amount of tax credits under the state homestead property tax credit.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mi/tax/income/credits/homestead_property_tax_credit/max_amount.yaml:1]:
```suggestion
description: Michigan provides this maximum amount under the state homestead property tax credit.
```

**Review by PavelMakarchuk — COMMENTED** [review]
.

File: policyengine_us/parameters/gov/states/mi/tax/income/credits/homestead_property_tax_credit/max_amount.yaml [hunk 9]
```diff
@@ -0,0 +1,10 @@
+description: Michigan provides this maximum amount under the state homestead property tax credit.
+metadata:
+    unit: currency-USD
+    period: year
+    label: Homestead Property Tax Credit max amount
+    reference:
+    - title: "Michigan Homestead Property Tax Credit"
+      href: "https://michiganfreetaxhelp.org/tax-credits/#:~:text=The%20Homestead%20Property%20Tax%20Credit,must%20be%20%24143%2C000%20or%20less."
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mi/tax/income/credits/homestead_property_tax_credit/max_amount.yaml:8]: need legal code

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: need to add [senior and disabled calculation](https://www.michigan.gov/taxes/-/media/Project/Websites/taxes/Forms/2022/2022-IIT-Forms/MI-1040CR.pdf?rev=3960a0f9b1de4f999acf1bb4209fc8dc&hash=956AD0168F04E84CF9841653AF0523A0)

Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: need to add [23% rent cap](http://www.legislature.mi.gov/(S(4kisbxgqhgmxdyusq0z31zad))/mileg.aspx?page=getObject&objectName=mcl-206-520)

Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: [legal](http://legislature.mi.gov/doc.aspx?mcl-206-30)

**Comment by MaxGhenis** [comment]
Looks like #2668 supersedes this

**Pull request closed without merging**